File: LondonDataServices.IDecide.Portal.Server/Models/PatientSearches/SearchCriteria.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

namespace LondonDataServices.IDecide.Portal.Server.Models.PatientSearches
{
    public class SearchCriteria
    {
        public string FirstName { get; set; } = string.Empty;
        public string Surname { get; set; } = string.Empty;
        public string Gender { get; set; } = string.Empty;
        public string Postcode { get; set; } = string.Empty;
        public string DateOfBirth { get; set; } = string.Empty;
        public string DateOfDeath { get; set; } = string.Empty;
        public string RegisteredGpPractice { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string PhoneNumber { get; set; } = string.Empty;
    }
}

File: LondonDataServices.IDecide.Manage.Client/src/models/searchCriterias/searchCriteria.ts
export class SearchCriteria {
    public nhsNumber: string = "";
    public firstName: string = "";
    public surname: string = "";
    public gender: string = "";
    public postcode: string = "";
    public dateOfBirth: string = "";
    public dateOfDeath: string = "";
    public registeredGpPractice: string = "";
    public email: string = "";
    public phoneNumber: string = "";

    constructor(searchCriteria?: Partial<SearchCriteria>) {
        if (searchCriteria) {
            this.nhsNumber = searchCriteria.nhsNumber ?? "";
            this.firstName = searchCriteria.firstName ?? "";
            this.surname = searchCriteria.surname ?? "";
            this.gender = searchCriteria.gender ?? "";
            this.postcode = searchCriteria.postcode ?? "";
            this.dateOfBirth = searchCriteria.dateOfBirth ?? "";
            this.dateOfDeath = searchCriteria.dateOfDeath ?? "";
            this.registeredGpPractice = searchCriteria.registeredGpPractice ?? "";
            this.email = searchCriteria.email ?? "";
            this.phoneNumber = searchCriteria.phoneNumber ?? "";
        }
    }
}